atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/AtlasConsoleJoins.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.console;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Maps;
+
+/**
+ * Created by dcohen on 4/17/17.
+ */
+public class AtlasConsoleJoins {
+
+    public enum JoinComponent {
+        JOIN_KEY, INPUT_VALUE, OUTPUT_VALUE
+    }
+
+    public static Iterable<Map<String, Object>> join(Iterable<Map<?, ?>> input, int batchSize,
+            Function<List<?>, List<Map<String, ?>>> getRowsFunction) {
+        Iterable<Map.Entry<?, ?>> entries = Iterables.concat(Iterables.transform(input, java.util.Map::entrySet));
+        return FluentIterable.from(Iterables.partition(entries, batchSize)).transformAndConcat(
+                batch -> {
+                    Map<?, ?> batchMap = batch.stream().collect(Collectors.toMap(e -> e.getKey(), e -> e.getValue()));
+                    List<?> keys = batch.stream().map(Map.Entry::getKey).collect(Collectors.toList());
```

Comment:
Oh, I see.  Table#getRows is confused by this because it expects a List as well and has this hacky listify(obj) method that assumes obj is a List, otherwise assumes it is a single element to be wrapped in a list.  Either way, eventually we'll have to materialize this this batch of keys to do the fetch, so I don't think this is really an important savings.